Cubical/Categories/Functor/Base.agda

```diff
@@ -0,0 +1,88 @@
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+
+module Cubical.Categories.Functor.Base where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Data.Sigma
+open import Cubical.Categories.Category
+
+private
+  variable
+    ℓC ℓC' ℓD ℓD' : Level
+
+record Functor (C : Precategory ℓC ℓC') (D : Precategory ℓD ℓD') : Type (ℓ-max (ℓ-max ℓC ℓC') (ℓ-max ℓD ℓD')) where
+  no-eta-equality
+  open Precategory
+
+  field
+    F-ob : C .ob → D .ob
+    F-hom : {x y : C .ob} → C [ x , y ] → D [(F-ob x) , (F-ob y)]
+    F-id : {x : C .ob} → F-hom (C .id x) ≡ D .id (F-ob x)
+    F-seq : {x y z : C .ob} (f : C [ x , y ]) (g : C [ y , z ]) → F-hom (f ⋆⟨ C ⟩ g) ≡ (F-hom f) ⋆⟨ D ⟩ (F-hom g)
+
+  isFull = (x y : _) (F[f] : D [(F-ob x) , (F-ob y)]) → ∃ (C [ x , y ]) (λ f → F-hom f ≡ F[f])
+  isFaithful = (x y : _) (f g : C [ x , y ]) → F-hom f ≡ F-hom g → f ≡ g
+  isEssentiallySurj = ∀ (d : D .ob) → Σ[ c ∈ C .ob ] CatIso {C = D} (F-ob c) d
+
+private
+  variable
+    ℓ ℓ' ℓ'' : Level
+    ℓE ℓE' : Level
+    B : Precategory ℓC ℓC'
+    C : Precategory ℓC ℓC'
+    D : Precategory ℓD ℓD'
+    E : Precategory ℓE ℓE'
+
```

Comment:
You don't need to declare `B C D E` at different levels, they will get generalized anyway.

Reply:
Noted